config/openssl.mpb

```diff
@@ -36,6 +38,15 @@ feature(openssl) {
   specific(prop:microsoft) {
     libpaths += $(SSL_ROOT)/lib/VC
   }
+
+  specific(cmake) {
```

Comment:
What if this did something like...
```
verbatim(cmake, foo, 1) {
  find_package(OpenSSL)
}
```

Reply:
It could.  But, I think the problem there is that if you want to build your code without OpenSSL, the find_package() would cause cmake to choke (assuming it wasn't available).  I am expecting things like that to be done at a higher level than the project level.

Reply:
There's an option of `find_package(OpenSSL REQUIRED)` -- not sure how that would be represented (if we wanted to), maybe it would relate to MPC features.

I don't mind leaving it out.  Just curious if it was easy to automate some of this...  I wonder what happens if CMake sees the same `find_package` multiple times?  Does it ignore the additional ones?

Reply:
Did you mean OPTIONAL?  Anyway, if it's there, having the valued for includes and libraries could cause a problem if the user is using something else to control whether OpenSSL is used or not.

It does appear to ignore duplicated find_package() calls.

Reply:
> I am expecting things like that to be done at a higher level than the project level.

Yes, that's the typical pattern, for them to be in the top level CMakeLists or something that file includes.

> Did you mean OPTIONAL?  Anyway, if it's there, having the valued for includes and libraries could cause a problem if the user is using something else to control whether OpenSSL is used or not.
> It does appear to ignore duplicated find_package() calls.

I don't think there's an explicit `OPTIONAL` argument, only an explicit `REQUIRED`. Also, yes additional calls should be ignored, but it's up to the package. I know ours used to throw a fatal error if called multiple times. Things that are builtin like OpenSSL should be safe though if we're not trying to force something that's different than what was declared before.